File: Dante1/Module1.09/Zadanie9.01/vector_utils.rs
pub fn read_vector(array: &mut [i32], size: usize, stop_value: i32) -> i32
{
    if size < 1 || size > 150
    {
        return -1;
    }

    let mut input: String = String::new();
    let mut counter: i32 = 0;

    for _i in 0..size
    {
        input.clear();
        std::io::stdin().read_line(&mut input).expect("Incorrect input");
        let number: i32 = input.trim().parse().expect("Incorrect parsing");

        if number == stop_value
        {
            break;
        }

        array[counter as usize] = number;
        counter += 1;
    }

    counter
}

pub fn display_vector(array: & [i32], size: usize)
{
    if size <= 0
    {
        return;
    }

    for i in 0..size
    {
        print!("{} ", array[i]);
    }
    println!();
}

pub fn bubble_sort(array: &mut [i32], size: usize) -> i32
{
    if size <= 0
    {
        return 1;
    }

    for _i in 0..size - 1
    {
        let mut pointer: u8 = 0;
        for j in 0..size - 1
        {
            if array[j] > array[j + 1]
            {
                (array[j], array[j + 1]) = (array[j + 1], array[j]);
                pointer = 1;
            }
        }
        display_vector(& array, size);
        if pointer == 0
        {
            break;
        }
    }

    0
}

File: Dante1/Module1.09/Zadanie9.01/vector_utils.c
#include <stdlib.h>
#include <stdio.h>
#include "vector_utils.h"

int read_vector(int *tab, int size, int stop_value)
{
    if (size < 1 || tab == NULL || size > 150)
    {
        return -1;
    }

    int counter = 0;
    for (int i = 0; i < size; i++)
    {
        int check = scanf("%i", &*(tab + i));
        if (check != 1)
        {
            return -2;
        }
        if (*(tab + i) == stop_value)
        {
            break;
        }
        counter++;
    }

    return counter;
}

void display_vector(const int* tab, int size)
{
    if(tab == NULL || size <= 0)
    {
        return;
    }
    for(int i = 0; i < size; i++)
    {
        printf("%i ",*(tab + i));
    }
    printf("\n");
}

int bubble_sort(int tab[], int size)
{
    if(tab == NULL || size <= 0)
    {
        return 1;
    }

    for(int i = 0; i < size - 1; i++)
    {
        int pointer = 0;
        for(int j = 0; j < size - 1; j++)
        {
            if(tab[j] > tab[j + 1])
            {
                int temp = tab[j];
                tab[j] = tab[j + 1];
                tab[j + 1] = temp;
                pointer = 1;
            }
        }
        display_vector(tab, size);
        if(pointer == 0)
        {
            break;
        }
    }

    return 0;
}
